arrow_pd_parser/parse/pandas_parser.py

```diff
@@ -0,0 +1,350 @@
+        if c["name"] not in df.columns:
+            raise ValueError(f"Column '{c['name']}' not in df")
+
+        else:
+            df[c["name"]] = cast_pandas_column_to_schema(
+                df[c["name"]],
+                metacol=c,
+                pd_integer=pd_integer,
+                pd_string=pd_string,
+                pd_boolean=pd_boolean,
+                pd_date_type=pd_date_type,
+                pd_timestamp_type=pd_timestamp_type,
+                num_errors=num_errors,
+                bool_map=bool_map,
+            )
+
+    final_cols = [c["name"] for c in meta["columns"] if c["name"] not in drop_columns]
+    df = df[final_cols]
+
+    return df
+
+
+def pd_read_csv(
+    input_file: Union[IO, str],
+    metadata: Union[Metadata, dict],
+    ignore_columns: List = None,
+    drop_columns: List = None,
+    pd_integer: bool = True,
+    pd_string: bool = True,
+    pd_boolean: bool = True,
+    pd_date_type: str = "datetime_object",
+    pd_timestamp_type: str = "datetime_object",
+    bool_map=None,
+    **kwargs,
+) -> pd.DataFrame:
+    """Read a csv file into a Pandas dataframe casting cols based on Metadata.
+
+    Args:
+        input_file (Union[IO, str]): the CSV you want to read. string, path or
+            file-like object.
+        metadata Union[Metadata, dict]: what you want the column to be cast to.
+        ignore_columns: (List, optional): a list of column names to not cast to the meta data dictionary.
+            These columns are remained unchanged.
+        drop_columns:  (List, optional): a list of column names you want to drop from the dataframe.
+        pd_boolean: whether to use the new pandas boolean format. Defaults to True.
+            When set to False, uses a custom boolean format to coerce object type.
+        pd_integer: if True, converts integers to Pandas int64 format.
+            If False, uses float64. Defaults to True.
+        pd_string: Defaults to True.
+        pd_date_type (str, optional): specifies the timestamp type. Can be one of:
+            "datetime_object", "pd_timestamp" or "pd_period" ("pd_period" not yet implemented).
+        pd_timestamp_type (str, optional): specifies the timestamp type. Can be one of:
+            "datetime_object", "pd_timestamp" or "pd_period" ("pd_period" not yet implemented).
+        bool_map (Callable, dict, optional): A custom mapping function that is applied to str
+            cols to be converted to booleans before conversion to boolean type.
+            e.g. {"Yes": True, "No": False}. If not set bool values are inferred by the
+            _default_str_bool_mapper.
+        **kwargs (optional): Additional kwargs are passed to pandas.read_csv
+
+    Returns:
+        Pandas DataFrame: the csv data as a dataframe, with the specified data types
+    """
+    if "low_memory" not in kwargs:
+        kwargs["low_memory"] = False
+    if "dtype" not in kwargs:
+        kwargs["dtype"] = str
+
+    df = pd.read_csv(input_file, **kwargs)
```

Comment:
Wouldn't you want this to have dtype as string by default?

Reply:
You mean instead of `str`?